mlir/test/Catalyst/ScatterTest.mlir

```diff
@@ -0,0 +1,181 @@
+    ^bb0(%arg2: tensor<i64>, %arg3: tensor<i64>):
+      %extracted_1 = tensor.extract %arg2[] : tensor<i64>
+      %extracted_2 = tensor.extract %arg3[] : tensor<i64>
+      %1 = arith.addi %extracted_1, %extracted_2 : i64
+      %from_elements = tensor.from_elements %1 : tensor<i64>
+      mhlo.return %from_elements : tensor<i64>
+  }) {
+    scatter_dimension_numbers = #mhlo.scatter<
+      update_window_dims = [2, 3],
+      inserted_window_dims = [0],
+      scatter_dims_to_operand_dims = [0, 1],
+      index_vector_dim = 2>,
+    indices_are_sorted = true,
+    unique_indices = true
+  } : (tensor<3x4x2xi64>, tensor<2x3x2xi32>, tensor<2x3x2x2xi64>) -> tensor<3x4x2xi64>
+  return %result : tensor<3x4x2xi64>
+}
+
+// CHECK: func.func private @__catalyst_update_scatter[[NUMBER0:.*]](%arg0: tensor<i64>, %arg1: tensor<i64>) -> tensor<i64>
+    // CHECK-NEXT:   [[EXTRACTED0:%.+]] = tensor.extract %arg0[] : tensor<i64>
+    // CHECK-NEXT:   [[EXTRACTED1:%.+]] = tensor.extract %arg1[] : tensor<i64>
+    // CHECK-NEXT:   [[RES:%.+]] = arith.addi [[EXTRACTED0]], [[EXTRACTED1]] : i64
+    // CHECK-NEXT:   [[RESTENSOR:%.+]] = tensor.from_elements [[RES]] : tensor<i64>
+    // CHECK-NEXT:   return [[RESTENSOR]] : tensor<i64>
+
+// CHECK: [[CALL1:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL2:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL3:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL4:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL5:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL6:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL7:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL8:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL9:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL10:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL11:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL12:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL13:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL14:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL15:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL16:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL17:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL18:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL19:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL20:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL21:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL22:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL23:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
+// CHECK: [[CALL24:%.+]] = call @__catalyst_update_scatter[[NUMBER0]](
```

Comment:
Should we check that there are not more than 24 calls? Also, is this supposed to be inside of a for-loop?

Reply:
Yes thanks!

Reply:
I will replace the for loop with a scf for loop as part of the technical debt